azkaban-common/src/main/java/azkaban/executor/AbstractExecutorManagerAdapter.java

```diff
@@ -0,0 +1,450 @@
+          this.apiGateway.callWithReference(pair.getFirst(), ConnectorParams.LOG_ACTION,
+              typeParam, jobIdParam, offsetParam, lengthParam, attemptParam);
+      return LogData.createLogDataFromObject(result);
+    } else {
+      return this.executorLoader.fetchLogs(exFlow.getExecutionId(), jobId, attempt,
+          offset, length);
+    }
+  }
+
+  protected List<Object> getExecutionJobStats(
+      final ExecutableFlow exFlow, final String jobId, final int attempt,
+      final Pair<ExecutionReference, ExecutableFlow> pair) throws ExecutorManagerException {
+    if (pair == null) {
+      return this.executorLoader.fetchAttachments(exFlow.getExecutionId(), jobId,
+          attempt);
+    }
+
+    final Pair<String, String> jobIdParam = new Pair<>("jobId", jobId);
+    final Pair<String, String> attemptParam =
+        new Pair<>("attempt", String.valueOf(attempt));
+
+    @SuppressWarnings("unchecked") final Map<String, Object> result =
+        this.apiGateway.callWithReference(pair.getFirst(), ConnectorParams.ATTACHMENTS_ACTION,
+            jobIdParam, attemptParam);
+
+    @SuppressWarnings("unchecked") final List<Object> jobStats = (List<Object>) result
+        .get("attachments");
+
+    return jobStats;
+  }
+
+  protected Map<String, Object> modifyExecutingJobs(final ExecutableFlow exFlow,
+      final String command,
+      final String userId, final Pair<ExecutionReference, ExecutableFlow> pair,
+      final String[] jobIds)
+      throws ExecutorManagerException {
+    if (pair == null) {
+      throw new ExecutorManagerException("Execution "
+          + exFlow.getExecutionId() + " of flow " + exFlow.getFlowId()
+          + " isn't running.");
+    }
+
+    final Map<String, Object> response;
+    if (jobIds != null && jobIds.length > 0) {
+      for (final String jobId : jobIds) {
+        if (!jobId.isEmpty()) {
+          final ExecutableNode node = exFlow.getExecutableNode(jobId);
+          if (node == null) {
+            throw new ExecutorManagerException("Job " + jobId
+                + " doesn't exist in execution " + exFlow.getExecutionId()
+                + ".");
+          }
+        }
+      }
+      final String ids = StringUtils.join(jobIds, ',');
+      response =
+          this.apiGateway.callWithReferenceByUser(pair.getFirst(),
+              ConnectorParams.MODIFY_EXECUTION_ACTION, userId,
+              new Pair<>(
+                  ConnectorParams.MODIFY_EXECUTION_ACTION_TYPE, command),
+              new Pair<>(ConnectorParams.MODIFY_JOBS_LIST, ids));
+    } else {
+      response =
+          this.apiGateway.callWithReferenceByUser(pair.getFirst(),
+              ConnectorParams.MODIFY_EXECUTION_ACTION, userId,
+              new Pair<>(
+                  ConnectorParams.MODIFY_EXECUTION_ACTION_TYPE, command));
+    }
+    return response;
+  }
+
+  /* Helper method for getRunningFlows */
+  protected List<Integer> getRunningFlowsHelper(final int projectId, final String flowId,
```

Comment:
Thanks for working on this refactoring!
Can this method be moved to `ExecutionControllerUtils` or `ExecutorUtils` class as an static method? Also, is it possible to merge those 2 classes?

Reply:
Sure. I have moved getRunningFlowsHelper  to ExecutorUtils. ExecutorUtils is for helper methods for executor related code. But ExecutionControllerUtils is related to executions. It is better to keep it separate.

Reply:
The 3 methods inside `ExecutorUtils` are used both in `ExecutionController` and `ExecutorManager` classes therefore I thought about merging the 2 utility classes.

Reply:
Aren't `ExecutionControllerUtils` and `ExecutorUtils` just utility classes for `ExecutionController` and `ExecutorManager` respectively?

Reply:
Not really. ExecutorUtils is for utilities related to executors so it is used by both ExecutionController and ExecutorManager.
ExecutionControllerUtils is more like utilities for executions and it is used by FlowRunner in executor.